framework/src/base/EigenProblem.C

```diff
@@ -124,6 +124,15 @@ EigenProblem::computeJacobian(const NumericVector<Number> & soln, SparseMatrix<N
   FEProblemBase::computeJacobian(soln, jacobian, kernel_type);
 }
 
+void
+EigenProblem::checkProblemIntegrity()
+{
+  FEProblemBase::checkProblemIntegrity();
+#if LIBMESH_HAVE_SLEPC
```

Comment:
We can keep this in for right now, but we need to fix the `#ifdef` business for SLEPc, because it is getting out of hands. A concept that worked for me in the past when doing a similar thing was to have no `#ifdef`s in header files, so that the interface was just one (as opposed to 2 or more versions based on how many ifdefs were there). Then the `#ifdef`s were in the .C where each section dealt with the one case. That way it was clear if the package was available or not and therefore what was safe to call... And than the clients of the interface did not have to know if the feature was available or not, they could just call the API...